### PR TITLE
enh(platform): Use API to select metrics in virtual metrics configuration form 21.10.x

### PR DESCRIPTION
**Pull request by hyahiaoui-ext**
## Description

Replacing www/include/views/graphs/common/makeXML_ListMetrics.php by the select2 for metrics

Listed metrics are from host/service couple selected in form

Previews: 

Before:
image
After:
image

**Fixes** # MON-14359

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: features/bootstrap/VirtualMetricHandleContext.php [hunk 1]
```diff
@@ -22,9 +22,11 @@ public function iAddAVirtualMetric()
         $this->page = new MetricsConfigurationListingPage($this);
         $this->assertFind('css', 'a[class="btc bt_success"]')->click();
         $this->page = new MetricsConfigurationPage($this);
+        sleep(60);
```
Comment (jeremyjaouen) [inline features/bootstrap/VirtualMetricHandleContext.php]: is this sleep intended ?

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: features/bootstrap/VirtualMetricHandleContext.php (hunk 1 above)
Comment (kduret) [reply features/bootstrap/VirtualMetricHandleContext.php]: `spin` method must be used

Comment (hyahiaoui-ext) [reply features/bootstrap/VirtualMetricHandleContext.php]: Even tho i used it, it seems like the tests don't pass for some reasons.

Comment (hyahiaoui-ext) [reply features/bootstrap/VirtualMetricHandleContext.php]: ah no it works now, for 70 seconds

**Pull request closed without merging**